manager/dispatcher/nodes.go

```diff
@@ -97,14 +104,17 @@ func (s *nodeStore) CheckRateLimit(id string) error {
 func (s *nodeStore) Add(n *api.Node, expireFunc func()) *registeredNode {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	var attempts int
 	if existRn, ok := s.nodes[n.ID]; ok {
+		attempts = existRn.Attempts
```

Comment:
By propagating the old attempts counter but resetting `Registered` to `time.Now()`, a situation like this would hit the rate limit:

- Register at t = 15
- Register again at t = 30
- Register again at t = 45
- Register again at t = 60

Is that the intended behavior?